pkg/gatewayserver/io/mqtt/format.go

```diff
@@ -25,7 +25,7 @@ type Format interface {
 	topics.Layout
 
 	FromDownlink(down *ttnpb.DownlinkMessage) ([]byte, error)
-	ToUplink(message []byte) (*ttnpb.UplinkMessage, error)
+	ToUplink(message []byte, gatewayID string) (*ttnpb.UplinkMessage, error)
```

Comment:
- `ids ttnpb.GatewayIdentifiers`
- add to others as well, although they're noop

Reply:
Second point here not addressed

Reply:
Fixed now - didn't understand the request initially.